pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -613,3 +658,17 @@ func DeleteFinalizer(object metav1.Object, finalizer string) {
 	finalizers.Delete(finalizer)
 	object.SetFinalizers(finalizers.List())
 }
+
+func buildRemoteClusterAPIClient(secretData string) (client.Client, error) {
```

Comment:
We should probably get this refactored out to a common utility as this is probably the second usage.